packages/back-end/src/queryRunners/QueryRunner.ts

```diff
@@ -191,13 +213,77 @@ export abstract class QueryRunner<
     });
   }
 
+  public async startReadyQueries(): Promise<void> {
+    const queuedQueries = await getQueriesByIds(
+      this.model.organization,
+      this.model.queries
+        .filter((q) => q.status === "queued")
+        .map((q) => q.query)
+    );
```

Comment:
We're already fetching all of the query objects in `refreshQueryStatuses()`.  We should just pass in the `queryMap` as an argument to this function.

Reply:
The solution I settled on feels a little hacky, since `refreshQueryStatuses()` doesn't really seem like it should return `QueryMap`, but I still prefer it to combining the methods since I feel like it improves the legibility of `onQueryFinish()`.

Open to other suggestions here.

Reply:
I think your solution is fine